### PR TITLE
RHCLOUD-34830 | feature: generate accurate metrics for the recipients resolver

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
By reporting the total requests, the successes and the failures, we will be able to set up some alerts for when the external services we depend on start returning unsuccessful responses.

## Jira ticket
[[RHCLOUD-34830]](https://issues.redhat.com/browse/RHCLOUD-34830)

### CODE REVIEW

File: recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/FetchUsersFromExternalServices.java [hunk 1]
```diff
@@ -108,27 +107,48 @@ public List<User> getUsers(String orgId, boolean adminsOnly) {
...
+            } else if (recipientsResolverConfig.isFetchUsersWithMbopEnabled()) {
+                Log.debug("Fetching users with BOP/MBOP");
+                supplier = "BOP";
+                users = fetchUsersWithMbop(orgId, adminsOnly);
+            } else {
+                Log.debug("Fetching users with IT service");
+                supplier = "IT";
+                users = fetchUsersWithItUserService(orgId, adminsOnly);
+            }
+        } catch (final FailsafeException | WebApplicationException e) {
```
Comment (MikelAlejoBR) [inline recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/FetchUsersFromExternalServices.java:135]: The only thing I changed about this bit is surrounding it with a `try / catch` block.

File: recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/FetchUsersFromExternalServices.java [hunk 2]
```diff
@@ -52,6 +53,13 @@
 @ApplicationScoped
 public class FetchUsersFromExternalServices {
 
+    public static final String COUNTER_FAILURES = "notifications.recipients-resolver.failures";
+    public static final String COUNTER_SUCCESSES = "notifications.recipients-resolver.successes";
+    protected static final String COUNTER_TAG_USER_PROVIDER = "user-provider";
+    protected static final String COUNTER_TAG_USER_PROVIDER_RBAC = "rbac";
+    protected static final String COUNTER_TAG_USER_PROVIDER_MBOP = "mbop";
+    protected static final String COUNTER_TAG_USER_PROVIDER_IT = "it";
+
```
Comment (MikelAlejoBR) [inline recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/FetchUsersFromExternalServices.java:62]: I wonder if we should just have a single `notifications.recipients-resolver.requests` counter and a `successful` or `failed` tags, along with the already implemented `user-provider` tag :thinking:

**Review by g-duval — APPROVED** [review]
LGTM

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**